include/exadg/structure/material/material.h

```diff
@@ -41,15 +41,28 @@ class Material
   {
   }
 
+  /*
+   * Evaluate 2nd Piola-Kirchhoff stress tensor given the gradient of the displacement field
+   * with respect to the reference configuration (not to be confused with the deformation gradient)
+   */
   virtual dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  evaluate_stress(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & E,
-                  unsigned int const                                              cell,
-                  unsigned int const                                              q) const = 0;
+  second_piola_kirchhoff_stress(
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & gradient_displacement,
+    unsigned int const                                              cell,
+    unsigned int const                                              q) const = 0;
 
+  /*
+   * Evaluate the directional derivative with respect to the displacement of the 2nd Piola-Kirchhoff
+   * stress tensor given the displacement increment shape function gradient *gradient_increment*
```

Comment:
I think one should better remove "shape function" here, because the argument is already the interpolation of shape functions into a q-point. 

So I would write `gradient of the displacment increment, short "gradient_increment"`. 

A minor thing: I would add "." at the end of the comments.